### ISSUE TITLE
TBL rules lack file and table filtering options

### ISSUE THREAD

**Issue by nozomi-koborinai** [nozomi-koborinai/contextlint#16]
## Problem

TBL-001 through TBL-004 are applied to all matched files and (in the case of TBL-001) all tables within those files. This causes false positives when a project has multiple types of tables or when rules should only apply to specific files.

### TBL-001: no table filtering

TBL-001 checks that required columns exist, but it applies to **every** table in every matched file. If a file contains both a requirements table and an unrelated table (e.g. a glossary), the unrelated table will be flagged:

```md
<!-- requirements.md -->
| ID | Status | Description |   ← OK
| ...

| Term | Definition |             ← ERROR: Missing "ID", "Status"
| ...
```

TBL-002, TBL-003, and TBL-004 naturally skip tables that don't have the specified column, so they don't have this particular issue.

### TBL-001–004: no `files` filter

Unlike SEC-001 (which has an optional `files` glob) and TBL-006 (which requires a `files` glob), TBL-001 through TBL-004 have no way to restrict which files they apply to. For example, there's no way to enforce an ID pattern only in `**/requirements.md` without also checking every other `.md` file.

## Proposal

Add an optional `files` glob parameter to TBL-001 through TBL-004, following the same pattern as SEC-001:

```jsonc
{ "rule": "tbl001", "options": { "requiredColumns": ["ID", "Status"], "files": "**/requirements.md" } }
```

When `files` is omitted, the rule applies to all files (current behavior, backwards-compatible).

This is the same approach used by SEC-001 (`files` option added in #6).
